common/persistence/sql/sqlplugin/mysql/nexus_incoming_services.go

```diff
@@ -25,47 +25,127 @@ package mysql
 import (
 	"context"
 	"database/sql"
+	"errors"
 
-	"go.temporal.io/api/serviceerror"
-
+	"go.temporal.io/server/common/persistence"
 	"go.temporal.io/server/common/persistence/sql/sqlplugin"
 )
 
+const (
+	createIncomingServicesTableVersionQry    = `INSERT INTO nexus_services_table_versions(service_type, version) VALUES(0, 1)`
+	incrementIncomingServicesTableVersionQry = `UPDATE nexus_services_table_versions SET version = ? WHERE service_type = 0 AND version = ?`
+	getIncomingServicesTableVersionQry       = `SELECT version FROM nexus_services_table_versions WHERE service_type = 0 LIMIT 1`
```

Comment:
You're already querying with the primary key, no need to limit.
```suggestion
	getIncomingServicesTableVersionQry       = `SELECT version FROM nexus_services_table_versions WHERE service_type = 0`
```